File: shinshun-1/program1.c
#include <stdio.h>

int main(void)
{
	int n,i,p,q,r;
	int count = 0;
	printf("N=");
	scanf("%d",&n);
	for(i=2;n!=1;)
	{
		p = n/i;	count++;
		q = i*p;
		r = n-q;
		if(r!=0)
		{
			i++;
		}
		else
		{
			printf("%d\n",i);
			n = n/i;
		}
	}
	printf("%d\n",count);
	return 0;
}

File: shinshun-1/program1.rb
count = 0;
print "N="
n = gets.to_i
for i in 2..n
	break if n == 1
	p = (n/i).to_i; count+=1;
	q = i*p
	r = n-q
	next if r != 0
	puts i
	n = p
	redo
end
puts count
